static/js/zamboni/devhub.js

```diff
@@ -990,7 +990,10 @@ function initCatFields(delegate) {
             $("input", $misc).prop("checked", checkedLength <= 0);
         };
         var checkOther = function() {
-            $("input", $main).prop("checked", false).prop("disabled", false);
+          // Save Android categories
+          var $androidCat = $main.find('input[name="form-1-categories"]');
+          $("input", $main).prop("checked", false).prop("disabled", false);
+          $($androidCat+":checked").prop("checked", true).prop("disabled", false);
```

Comment:
@raajitr I think the next step is to more properly understand what root cause of the bug is and then figure out the necessary lookups to get what you need.